builder/anka/step_clone_vm.go

```diff
@@ -0,0 +1,306 @@
+package anka
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	"github.com/hashicorp/packer-plugin-sdk/packer"
+	"github.com/veertuinc/packer-builder-veertu-anka/client"
+	"github.com/veertuinc/packer-builder-veertu-anka/common"
+	"github.com/veertuinc/packer-builder-veertu-anka/util"
+)
+
+// StepCloneVM will be used to run the clone step for any 'vm-clone' builder types
+type StepCloneVM struct {
+	client client.Client
+	vmName string
+}
+
+// Run clones a vm from a source vm either from an anka registry or locally
+func (s *StepCloneVM) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	config := state.Get("config").(*Config)
+	ui := state.Get("ui").(packer.Ui)
+	util := state.Get("util").(util.Util)
+	onError := func(err error) multistep.StepAction {
+		return util.StepError(ui, state, err)
+	}
+	sourceVMTag := "latest"
+
+	if config.SourceVMTag != "" {
+		sourceVMTag = config.SourceVMTag
+	}
+
+	registryParams := client.RegistryParams{
+		RegistryName: config.RegistryName,
+		RegistryURL:  config.RegistryURL,
+		NodeCertPath: config.NodeCertPath,
+		NodeKeyPath:  config.NodeKeyPath,
+		CaRootPath:   config.CaRootPath,
+		IsInsecure:   config.IsInsecure,
+	}
+
+	s.client = state.Get("client").(client.Client)
+	s.vmName = config.VMName
+
+	state.Put("vm_name", s.vmName)
+
+	if config.PackerForce {
+		exists, err := s.client.Exists(s.vmName)
+		if err != nil {
+			return onError(err)
+		}
+		if exists {
+			ui.Say(fmt.Sprintf("Deleting existing virtual machine %s", s.vmName))
+
+			err = s.client.Delete(client.DeleteParams{VMName: s.vmName})
+			if err != nil {
+				return onError(err)
+			}
+		}
+	}
+
+	registryPullParams := client.RegistryPullParams{
```

Comment:
I might restructure this section a bit to clean it up. 

1. check if sourceExists
2. If !sourceExists OR config.AlwaysFetch — registry pull (create the params in this block)

Reply:
So my only rebuttal: why would we want to check if the source exists locally if they specify we pull it from the registry always?

Reply:
Fair. In my mind the extra call is worth avoiding the duplication of logic. I don't like having the same 5 lines repeated twice in both branches of logic. 

```
doPull = config.AlwaysFetch
if !config.AlwaysFetch { 
   check for source
   if source does not exist, set doPull to true
}

if doPull {
  pull from registry
}
```